jobs/ror/ProcessRorCsv.php

```diff
@@ -0,0 +1,287 @@
+<?php
+
+/**
+ * @file jobs/ror/ProcessRorCsv.php
+ *
+ * Copyright (c) 2014-2025 Simon Fraser University
+ * Copyright (c) 2000-2025 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class ProcessRorCsv
+ *
+ * @brief Job to import ROR dataset in chunks/batches
+ */
+
+namespace PKP\jobs\ror;
+
+use Throwable;
+use Illuminate\Queue\Middleware\WithoutOverlapping;
+use PKP\jobs\BaseJob;
+use Illuminate\Bus\Batchable;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Schema;
+use PKP\task\UpdateRorRegistryDataset;
+use Illuminate\Database\Schema\Blueprint;
+use PKP\scheduledTask\ScheduledTaskHelper;
+
+class ProcessRorCsv extends BaseJob
+{
+    /**
+     * Indicate if the job should be marked as failed on timeout.
+     */
+    public bool $failOnTimeout = false;
+
+    use Batchable;
+
+    public function __construct(
+        protected string $pathCsv,
+        protected int $startRow,
+        protected int $endRow,
+        protected array $dataMapping,
+        protected array $dataMappingIndex,
+        protected string $noLocale,
+        protected string $temporaryTable,
+        protected string $scheduledTaskLogFilePath
+    )
+    {
+        parent::__construct();
+    }
+
+    /**
+     * Get the middleware the job should pass through.
+     */
+    public function middleware()
+    {
+        return [
+            (new WithoutOverlapping("{$this->pathCsv}:{$this->startRow}-{$this->endRow}"))->expireAfter(60), // 1-minute lock
+        ];
+    }
+
+    public function handle()
+    {
+        try {
+            UpdateRorRegistryDataset::writeToExecutionLogFile(
+                "Importing batch chunk {$this->startRow}-{$this->endRow} starting",
+                $this->scheduledTaskLogFilePath,
+                ScheduledTaskHelper::SCHEDULED_TASK_MESSAGE_TYPE_NOTICE
+            );
+
+            $this->createTemporaryTable();
+
+            $batchRows = [];
+            $currentRow = 0;
+            $isHeader = true;
+
+            if (($handle = fopen($this->pathCsv, 'r')) !== false) {
+                while (($rowCsv = fgetcsv($handle, null, ',', '"', '\\')) !== false) {
+                    if ($isHeader) {
+                        foreach ($this->dataMapping as $keyDB => $keyCsv) {
+                            $this->dataMappingIndex[$keyDB] = array_search($keyCsv, $rowCsv, true);
+                        }
+                        $isHeader = false;
+                        continue;
+                    }
```

Comment:
Instead of doing this check on every row, you can move it out of the loop, and add an initial fetch, like `if (($header = fgetcsv($handle)) !=== false) { ... } while (...)`